pkg/charts/charts.go

```diff
@@ -175,3 +175,23 @@ func InsertVersionValues(values chartutil.Values) (chartutil.Values, error) {
 	}
 	return chartutil.ReadValues(InsertVersion([]byte(raw)))
 }
+
+// MergeMaps returns the resultant map after merging two maps
```

Comment:
Can we flesh out this godoc with more details about the semantics of this method?  It should answer at least these questions:
* Are the inputs mutated?
* Which input takes precedence?
* How does this differ from other merge functions such as the one in mergo?